libV2/schemaUtils.js

```diff
@@ -1949,7 +2060,33 @@ let QUERYPARAM = 'query',
         return;
       }
 
-      let paramValue = resolveValueOfParameter(context, param);
+      let propertyDetails = {},
+        properties = {},
+        keyName,
+        paramValue = resolveValueOfParameter(context, param);
+
+      if (param && param.schema) {
+        const { name, schema } = param;
+        keyName = name;
+        properties = {
+          type: schema.type || 'unknown',
+          format: schema.format || undefined,
+          default: schema.default || undefined,
+          required: param.required || false,
+          deprecated: param.deprecated || false,
+          enum: schema.enum || undefined,
+          minLength: schema.minLength || undefined,
+          maxLength: schema.maxLength || undefined,
+          minimum: schema.minimum || undefined,
+          maximum: schema.maximum || undefined,
+          pattern: schema.pattern || undefined,
+          example: schema.example || undefined
```

Comment:
Also this piece of code seems to have been used in multiple places. Can this be abstracted to a function and reused ?